src/network.c

```diff
@@ -1008,16 +1008,24 @@ int network_identifier_split(char* addr, char* ip_out, int* port_out) {
 	if((size_t)ip_start <= 6)
 		return 0;
 	char* port_start = strchr(ip_start, ':');
-	*port_out = port_start ? strtoul(port_start + 1, NULL, 10) : 32887;
 
-	if(strchr(ip_start, '.')) {
-		if(port_start) {
-			strncpy(ip_out, ip_start, port_start - ip_start);
-			ip_out[port_start - ip_start] = 0;
-		} else {
-			strcpy(ip_out, ip_start);
-		}
-	} else {
+	// parse port, also ignore anything past another ':' (e.g. version info)
+	int next_colon = strchr(port_start + 1, ':');
+	*port_out = strtoul(port_start + 1, next_colon, 10);
```

Comment:
next_colon should be passed as a `char**` pointer, where `strtoul` can write the first invalid char pos to, HOWEVER we don't actually need it here, and can pass `NULL` instead, thus don't even need it in the first place

Reply:
does this PR even do anything different to the original code, except the additional ip length check???